airflow/models/dagrun.py

```diff
@@ -838,14 +838,24 @@ def verify_integrity(self, session: Session = NEW_SESSION):
                     ti.state = State.REMOVED
                 continue
 
-            if task.is_mapped:
-                task = cast("MappedOperator", task)
-                num_mapped_tis = task.parse_time_mapped_ti_count
-                # Check if the number of mapped literals has changed and we need to mark this TI as removed
-                if not num_mapped_tis or ti.map_index >= num_mapped_tis:
+            if not task.is_mapped:
+                continue
+            task = cast("MappedOperator", task)
```

Comment:
nit: Should we extract L843-858 to a separate function? This might help test that function separately too